### PR TITLE
Ben instructor search page

### LINKED ISSUE #2: FEATURE: Add search by instructor(menu item takes you to a form)
The "Search by Instructor" menu item takes you to a form where you can select a start quarter, end quarter, and input an instructor name.

DO THIS AFTER ISSUE #1 

Implementation Todos
Frontend:

 There will be a new page for the search (with storybook entries and tests)
Backend:

 There will be a new backend endpoint to support this search (with tests)

### PR DESCRIPTION
**Pull request by BenCrutt**
Added form for instructor search, including dropdown selects for start quarter and end quarter. Then adds a text box to input the course instructor name for search. Currently, submit button does nothing, only frontend is built. Later hitting submit will show a table of courses that fall into the criteria given in the form. Resolves #2 

StoryBook:
Before(No Story File)
[After](https://ucsb-cs156-s23.github.io/proj-courses-s23-7pm-1/prs/36/storybook/?path=/docs/pages-courseinstructorindexpage--default)

Screenshot:
Before:
Screenshot 2023-05-30 at 2 41 24 PM

After:
Screenshot 2023-06-02 215639

### CODE REVIEW

**Review by kdhulipala41 — DISMISSED** [review]
LGTM

File: frontend/src/main/components/Nav/AppNavbar.js [hunk 1]
```diff
@@ -84,7 +84,11 @@ export default function AppNavbar({ currentUser, systemInfo, doLogout, currentUr
               </NavDropdown>
             </Nav>
 
-
+            <Nav className="mr-auto">
+              <NavDropdown title="Course Instructor" id="appnavbar-course-instructor-dropdown" data-testid="appnavbar-course-instructor-dropdown" >
+                <NavDropdown.Item href="/courseinstructor/search" data-testid="appnavbar-course-instructor-search">Search</NavDropdown.Item>
+              </NavDropdown>
+            </Nav>
```
Comment (pconrad) [inline frontend/src/main/components/Nav/AppNavbar.js]: If a dropdown has only one item in it, why even make it a drop down?  Why not just make it a link?

See https://react-bootstrap.github.io/docs/components/navbar for options and examples.

Also: see my feedback on #29 ... I'm just not sold on "Course Instructor" as a helpful label for this activity.  Maybe you can convince me, but I think "search by instructor" or even "Search by" with instructor / location / course number as the things in the dropdown menu would make much more sense...

File: frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { useState } from "react";
+import { Form, Button, Container, Row, Col } from "react-bootstrap";
+
+//import { allTheLevels } from "fixtures/levelsFixtures";
+import { quarterRange } from "main/utils/quarterUtilities";
+
+import { useSystemInfo } from "main/utils/systemInfo";
+import SingleQuarterDropdown from "../Quarters/SingleQuarterDropdown";
+//import SingleSubjectDropdown from "../Subjects/SingleSubjectDropdown";
+//import SingleLevelDropdown from "../Levels/SingleLevelDropdown";
```
Comment (pconrad) [inline frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js]: We really try not to include commented out code in code that we are merging into main.

File: frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  const endQtr = systemInfo?.endQtrYYYYQ || "20214";
+  // Stryker enable OptionalChaining
+
+  const quarters = quarterRange(startQtr, endQtr);
+
+  // Stryker disable all : not sure how to test/mock local storage
+  const localStartQuarter = localStorage.getItem("InstructorCourseSearch.StartQuarter");
+  const localEndQuarter = localStorage.getItem("InstructorCourseSearch.EndQuarter");
+  //const localSubject = localStorage.getItem("InstructorCourseSearch.Subject");
+  //const localCourseNumber = localStorage.getItem("InstructorCourseSearch.CourseNumber");
```
Comment (pconrad) [inline frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js]: Do we need this commented out code?

File: frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    ["/api/UCSBSubjects/all"], 
+    { method: "GET", url: "/api/UCSBSubjects/all" }, 
+    []
+  );
+
+  const [startQuarter, setStartQuarter] = useState(localStartQuarter || quarters[0].yyyyq);
+  const [endQuarter, setEndQuarter] = useState(localEndQuarter || quarters[0].yyyyq);
+  //const [subject, setSubject] = useState(localSubject || {});
+  //const [courseNumber, setCourseNumber] = useState(localCourseNumber || "");
+  //const [courseSuf, setCourseSuf] = useState("");
```
Comment (pconrad) [inline frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js]: Do we need this?

File: frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  const [instructor, setInstructor] = useState(localInstructor || "");
+    
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    //fetchJSON(event, { startQuarter, endQuarter, subject, courseNumber, courseSuf });
```
Comment (pconrad) [inline frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js]: Do we need this commented out code

**Review by pconrad — CHANGES_REQUESTED** [review]
A few small things to take a look at

File: frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import { useBackend  } from "main/utils/useBackend";
+
+const InstructorCourseSearchForm = ({ fetchJSON }) => {
+
+  const { data: systemInfo } = useSystemInfo();
+
+  // Stryker disable OptionalChaining
+  const startQtr = systemInfo?.startQtrYYYYQ || "20211";
+  const endQtr = systemInfo?.endQtrYYYYQ || "20214";
+  // Stryker enable OptionalChaining
```
Comment (pconrad) [inline frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js]: Should be:
```suggestion
  // Stryker restore OptionalChaining
```

See: https://ucsb-cs156-s23.slack.com/archives/C050779M19P/p1685726928710669

File: frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  const [startQuarter, setStartQuarter] = useState(localStartQuarter || quarters[0].yyyyq);
+  const [endQuarter, setEndQuarter] = useState(localEndQuarter || quarters[0].yyyyq);
+  const [instructor, setInstructor] = useState(localInstructor || "");
+    
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    fetchJSON(event, { startQuarter, endQuarter, instructor});
+  };
+
+  // Stryker disable all : Stryker is testing by changing the padding to 0. But this is simply a visual optimization as it makes it look better
```
Comment (pconrad) [inline frontend/src/main/components/BasicCourseSearch/InstructorCourseSearchForm.js]: Please remove this.  We have a documented easy way to test this, so this is no longer an appropriate stryker exception.

See: https://ucsb-cs156.github.io/topics/stryker/stryker_html_attribute_surviving_mutants.html
See:

**Review by pconrad — CHANGES_REQUESTED** [review]
This mostly looks good! 

A few small things that should be changed (see below)

**Review by pconrad — APPROVED** [review]
LGTM!  Will test on staff-qa before merging

**Pull request merged**